Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -78,45 +92,49 @@ void preparePathForEditing() {
             // Open a new path that the contents of the old path will be streamed into. We cannot re-use the same path as it is now closed
             // and cannot be opened again. We use the newPath variable because the factory was returning the same pointer for some strange
             // reason so this will force it to do otherwise.
-            FAIL_FAST_IF_FAILED(factory->CreatePathGeometry(&newPath));
-            FAIL_FAST_IF_FAILED(newPath->Open(&newSink));
-            FAIL_FAST_IF_FAILED(_impl.pathGeometry->Stream(newSink.Get()));
+            RETURN_IF_FAILED(factory->CreatePathGeometry(&newPath));
+            RETURN_IF_FAILED(newPath->Open(&newSink));
+            RETURN_IF_FAILED(PathGeometry()->Stream(newSink.Get()));
 
-            _impl.pathGeometry = newPath;
-            _impl.geometrySink = newSink;
+            PathGeometry() = newPath;
+            GeometrySink() = newSink;
 
             // Without a new figure being created, it's by default closed
-            _impl.isFigureClosed = true;
+            _impl.figureClosed = true;
         }
+        return S_OK;
     }
 
-    void closePath() {
-        if (_impl.geometrySink) {
-            endFigure(D2D1_FIGURE_END_OPEN);
-            _impl.geometrySink->Close();
-            _impl.geometrySink = nullptr;
+    void ClosePath() {
+        if (GeometrySink()) {
+            EndFigure(D2D1_FIGURE_END_OPEN);
+            GeometrySink()->Close();
+            GeometrySink() = nullptr;
         }
     }
 
-    void beginFigure() {
-        if (_impl.isFigureClosed) {
-            _impl.geometrySink->BeginFigure(_CGPointToD2D_F(_impl.currentPoint), D2D1_FIGURE_BEGIN_FILLED);
-            _impl.isFigureClosed = false;
+    void BeginFigure() {
+        if (_impl.figureClosed) {
+            GeometrySink()->BeginFigure(_CGPointToD2D_F(CurrentPoint()), D2D1_FIGURE_BEGIN_FILLED);
+            _impl.figureClosed = false;
         }
```

Comment:
I agree this is getting out of control because it is not clear what pattern we are using here.  Although the impl struct is to work around inability to add things directly to outer struct, the entire thing fits into the proxy pattern.  From that perspective, forwarding all calls to the impl makes a lot of sense.  

The reason this feels like getting out of hand is because we are not encapsulating Impl the right way.  There is nothing wrong in writing plain old c code and access the struct members directly, but given we are in C++ land, we might as well be better about it.